controllers/lvm_volumegroup.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	lvmvgName = "lvmvg-manager"
+)
+
+type lvmVG struct{}
+
+func (c lvmVG) getName() string {
+	return lvmvgName
+}
+
+func (c lvmVG) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	lvmVolumeGroups, err := c.getLvmVolumeGroups(r, lvmCluster)
+	if err != nil {
+		return err
+	}
+
+	for _, volumeGroup := range lvmVolumeGroups {
+		result, err := cutil.CreateOrUpdate(ctx, r.Client, volumeGroup, func() error {
+			// no need to mutate any field
+			return nil
+		})
+
+		if err != nil {
+			r.Log.Error(err, "LvmVolumeGroup reconcile failure", "name", volumeGroup.Name)
+		} else {
+			r.Log.Info("LvmVolumeGroup", "operation", result, "name", volumeGroup.Name)
+		}
+
+	}
+	return nil
+}
+
+func (c lvmVG) ensureDeleted(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	existingLvmVg := &lvmv1alpha1.LVMVolumeGroup{}
+	vgcrs, err := c.getLvmVolumeGroups(r, lvmCluster)
+	if err != nil {
+		return err
+	}
+
+	for _, vgcr := range vgcrs {
+		err := r.Client.Get(ctx, types.NamespacedName{Name: vgcr.Name, Namespace: vgcr.Namespace}, existingLvmVg)
+
+		if err != nil {
+			// already deleted in previous reconcile
+			if errors.IsNotFound(err) {
+				r.Log.Info("LVMVG deleted", "Name", vgcr.Name)
+				return nil
+			}
+			r.Log.Error(err, "failed to retrieve LVMVG resource", "Name", vgcr.Name)
```

Comment:
":s/Name/name" in log lines

Reply:
The Name is the fieldname. That will be capitalized

Reply:
Its not consistent through out.  Both `name` and `Name` are used in the logs.

Reply:
Ack. Changing it to "name".